More content consistency, less undef content warning.

jeteve opened the pull request:
Hi,

This pull request addresses the case where some client code would call $m->content(undef);

This used to cause the '_content' property to go 'undef' instead of keeping the default 'empty string' value, and subsequently generating warnings on 'to_string' for instance.

This fixes this issue by enforcing more consistency and setting the content to be an empty string when one calls $m->content(undef).

Additionally, this introduces fixes for the test suite (it now fully passes straight away), and some travis building, checking that this package remains compatible with perl >= 5.8: https://travis-ci.org/jeteve/http-message

jeteve commented:
Hi @karenetheridge 

Any thought on that?

karenetheridge commented:
Since this is @gisle's code, I've only been shipping obvious bug fixes and tidy-ups to the distribution packaging.

jeteve commented:
Cool. Thx @karenetheridge :)

Hi @gisle do you mind reviewing this?

patmooney commented:
I'm also seeing warnings about content undef, so this has my vote!

ziali088 commented:
+1

bor commented:
+1

bor commented:
I got such warnings after update my old 6.06 version.
AFAIU those warnings was appears after 4fde9ad8bf077e7d0d3ea9247590f458d60aaaea

jeteve commented:
@karenetheridge @gisle , I consolidated my changes in one commit and removed the extra Test dependency as required.

I replaced it with a skip_all at test time, allowing successful automated run of the test suite: https://travis-ci.org/jeteve/http-message

I'd like to point out that #30 is a false good idea, cause it will have the unfortunate consequence of hidding the potential misbehaviour of some other code.

stuartskelton commented:
I have applied this change locally on my machine, as the warnings were getting a little annoying and the can report no adverse effects.

jeteve commented:
@karenetheridge @gisle any chance to merge this for the next release?

jeteve commented:
Hi @karenetheridge and @gisle . Any chance to get that merged for the new year?

Thanks!

Jerome.

1nickt commented:
++ for this fix, hoping for it to be merged please.

PopeFelix commented:
++ - it would remove the necessity for a some jiggery-pokery to avoid warnings.

jeteve commented:
Any thoughts?

oalders commented:
This looks good to me.  @jeteve could you fix the merge conflict?

oalders commented:
Closed via #37
